kube-core/src/object.rs

```diff
@@ -337,14 +344,53 @@ mod test {
         assert_eq!(mypod.namespace().unwrap(), "dev");
         assert_eq!(mypod.name_unchecked(), "blog");
         assert!(mypod.status().is_none());
-        assert_eq!(mypod.spec().containers[0], ContainerSimple {
-            image: "blog".into()
-        });
+        assert_eq!(
+            mypod.spec().containers[0],
+            ContainerSimple { image: "blog".into() }
+        );
 
         assert_eq!(PodSimple::api_version(&ar), "v1");
         assert_eq!(PodSimple::version(&ar), "v1");
         assert_eq!(PodSimple::plural(&ar), "pods");
         assert_eq!(PodSimple::kind(&ar), "Pod");
         assert_eq!(PodSimple::group(&ar), "");
     }
+
+    #[test]
+    fn k8s_object_list() {
+        use k8s_openapi::api::core::v1::Pod;
+        // by grabbing the ApiResource info from the Resource trait
+        let ar = ApiResource::erase::<Pod>(&());
+        assert_eq!(ar.group, "");
+        assert_eq!(ar.kind, "Pod");
+        let podlist: ObjectList<Pod> = ObjectList {
+            types: TypeMeta{
+                api_version: ar.api_version,
+                kind: ar.kind + "List",
+            },
```

Comment:
Maybe `TypeMeta` needs a constructor here like `TypeMeta::list_kind<K: Resource>()`?

Reply:
It may need one, but regularly the type meta is set by the API server on the returned object, and the client is not supposed to prepare this object directly.

Reply:
True, but it's nice to have. I like the ones you added and their docs. Just nit on naming below upon thinking about it.